search.go

```diff
@@ -0,0 +1,213 @@
+	default:
+		return nil, nil, fmt.Errorf("unexepected http status code %v", resp.StatusCode)
+	}
+}
+
+// HistogramBucket corresponds to the histogram display in the Sumo Logic interactive analytics API.
+type HistogramBucket struct {
+	Length         int `json:"length"`
+	Count          int `json:"count"`
+	StartTimeStamp int `json:"startTimeStamp"`
+}
+
+// SearchJobStatusResponse stores the response from getting a search status.
+type SearchJobStatusResponse struct {
+	State           string             `json:"state"`
+	MessageCount    int                `json:"messageCount"`
+	HistgramBuckets []*HistogramBucket `json:"histogramBuckets"`
+	RecordCount     int                `json:"recordCount"`
+	PendingWarnings []string           `json:"pendingWarnings"`
+	PendingErrors   []string           `json:"pendingErrors"`
+}
+
+// GetSearchJobStatus retrieves the status of a running job.
+func (c *Client) GetSearchJobStatus(searchJobID string, cookies []*http.Cookie) (*SearchJobStatusResponse, error) {
+
+	relativeURL, _ := url.Parse(fmt.Sprintf("search/jobs/%s", searchJobID))
+	url := c.EndpointURL.ResolveReference(relativeURL)
+	req, err := http.NewRequest("GET", url.String(), nil)
+	req.Header.Add("Content-Type", "application/json")
+	req.Header.Add("Authorization", "Basic "+c.AuthToken)
+	for _, v := range cookies {
+		req.AddCookie(v)
+	}
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	responseBody, _ := ioutil.ReadAll(resp.Body)
+
+	switch resp.StatusCode {
+	case http.StatusOK:
+		var jobStatus = new(SearchJobStatusResponse)
+		err = json.Unmarshal(responseBody, &jobStatus)
+		if err != nil {
+			return nil, err
+		}
+		return jobStatus, nil
+	default:
+		return nil, fmt.Errorf("Status not OK : %v", resp.StatusCode)
+	}
+}
+
+// SearchJobResultsRequest is a wrapper for the search job messages params.
+type SearchJobResultsRequest struct {
+	ID     string `json:"searchJobId"`
+	Offset int    `json:"offset"`
+	Limit  int    `json:"limit"`
+}
+
+// SearchJobResultField is one field from a search result.
+type SearchJobResultField struct {
+	Name      string `json:"name"`
+	FieldType string `json:"fieldType"`
+	KeyField  bool   `json:"keyField"`
+}
+
+// SearchJobResultMessage represents one message from a search job result.
+type SearchJobResultMessage struct {
+	// not 100% sure about this or if it should be map[string]interface{}, map[string]string or completely different approach.
+	// Depending on the origin of the log, the structure of this may vary.
+	// The thought is apps including this package will define a struct for the specific message types
+	// and parse the _raw field into that struct of for the app's use case.
+	Map map[string]interface{} `json:"map"`
+}
+
+// SearchJobResult represents a search job result
+type SearchJobResult struct {
+	Fields   []*SearchJobResultField   `json:"fields"`
```

Comment:
I wouldn't use a slice of pointers here without good reason. `[]SearchJobResultField` and similar for the next line.